src/agent/debuglet.ts

```diff
@@ -107,6 +113,41 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise stores a promise. This promise can be resolved by calling
+ * function resolve() and can only be resolved once.
+ */
+export class CachedPromise {
+  private promiseResolve: (() => void) | null;
```

Comment:
nit: Use `private promiseResolve: (() => void) | null = null` so that the initial value of `promiseResolve` is `null`.

Reply:
Done.